File: java/Fibonacci.java
/**
 * 
 * @author plim
 *
 * Simple Fibonacci number functions
 */
public class Fibonacci {
  public static int recursive(int n) {
    if (n <= 2) {
      return 1;
    } else {
      return recursive(n-1) + recursive(n-2);
    }
  }
  
  public static int interative(int n) {
    if (n <= 2) {
      return 1;
    }

    int first = 1;
    int second = 1;
    int third = 0;
    
    for (int i = 3; i <= n; i++) {
      third = first + second;
      first = second;
      second = third;
    }
    return third;
  }
  
  public static void main(String args[]) {
    // recursive
    System.out.println("Recursive Printout:");
    for (int i = 1; i <= 10; i++) {
      System.out.println(i + ": " + recursive(i));
    }
    
    // iterative
    System.out.println("Iterative Printout:");
    for (int i = 1; i <= 10; i++) {
      System.out.println(i + ": " + interative(i));
    }
  }
}

File: ruby/fibonacci.rb

class Fibonacci
  def self.recursive(n)
    if n <= 2
      return 1
    else
      return recursive(n-1) + recursive(n-2)
    end
  end
  
  def self.iterative(n)
    if n <= 2
      return 1
    end
    
    first = 1
    second = 1
    third = 0
    
    3.upto(n) do |i|
      third = first + second
      first = second
      second = third
    end
    
    return third
  end
end

# recursive
1.upto(10) do |n|
  puts Fibonacci.recursive(n)
end

# iterative
1.upto(10) do |n|
  puts Fibonacci.iterative(n)
end
